airbyte-integrations/connectors/source-surveymonkey/source_surveymonkey/streams.py

```diff
@@ -0,0 +1,265 @@
+        cursor_field: List[str] = None,
+        stream_slice: Mapping[str, Any] = None,
+        stream_state: Mapping[str, Any] = None,
+    ) -> Iterable[Mapping[str, Any]]:
+        with vcr.use_cassette("pretty_name.yaml", record_mode="new_episodes"):
+            yield from super().read_records(
+                sync_mode=sync_mode, cursor_field=cursor_field, stream_slice=stream_slice, stream_state=stream_state
+            )
+
+
+class IncrementalSurveymonkeyStream(SurveymonkeyStream, ABC):
+
+    state_checkpoint_interval = 1000
+
+    @property
+    @abstractmethod
+    def cursor_field(self) -> str:
+        """
+        Defining a cursor field indicates that a stream is incremental, so any incremental stream must extend this class
+        and define a cursor field.
+        """
+
+    def request_params(self, stream_state: Mapping[str, Any], **kwargs) -> MutableMapping[str, Any]:
+        params = super().request_params(stream_state=stream_state, **kwargs)
+        params["sort_order"] = "ASC"
+        params["sort_by"] = "date_modified"
+        params["per_page"] = 1000  # maybe as user input or bigger value
+        since_value = pendulum.parse(stream_state.get(self.cursor_field)) if stream_state.get(self.cursor_field) else self._start_date
+
+        since_value = max(since_value, self._start_date)
+        params["start_modified_at"] = since_value.strftime("%Y-%m-%dT%H:%M:%S")
+        return params
+
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
+        """
+        Return the latest state by comparing the cursor value in the latest record with the stream's most recent state object
+        and returning an updated state object.
+        """
+        latest_state = latest_record.get(self.cursor_field)
+        current_state = current_stream_state.get(self.cursor_field) or latest_state
+        if current_state:
+            return {self.cursor_field: max(latest_state, current_state)}
+        return {}
+
+
+class Surveys(IncrementalSurveymonkeyStream):
+    """
+    Docs: https://developer.surveymonkey.com/api/v3/#surveys
+    A source for stream slices. It does not contain useful info itself.
+    """
+
+    cursor_field = "date_modified"
+
+    def path(self, **kwargs) -> str:
+        return "surveys"
+
+    def parse_response(self, response: requests.Response, stream_state: Mapping[str, Any], **kwargs) -> Iterable[Mapping]:
+        response_json = response.json()
+        if response_json.get("error"):
+            raise Exception(response_json.get("error"))  # TODO: apply ShouldRetry
+        result = response_json.get(self.data_field, [])
+        for record in result:
+            substream = SurveyDetails(survey_id=record["id"], start_date=self._start_date, authenticator=self.authenticator)
+            child_record = substream.read_records(sync_mode=SyncMode.full_refresh)
+            yield from child_record
```

Comment:
Why do you sync child stream in parse_response?

Reply:
parent stream contains title, href, link only, details have expanded response including cursor_field. Parent is useless for data, it requires for enumeration.

Reply:
@vovavovavovavova I got it. But why did you put read in parse_response method?

Reply:
`/surveys` contains the following schema: 

````
{
  "title": "My Survey",
  "nickname": "",
  "category":"",
  "language": "en",
  "question_count": 10,
  "page_count": 10,
  "date_created": "2015-10-06T12:56:55",
  "date_modified": "2015-10-06T12:56:55",
  "id": "1234",
  "href": "https://api.surveymonkey.com/v3/surveys/1234",
  "buttons_text": {
    "done_button": "Done",
    "prev_button": "Prev",
    "exit_button": "Exit",
    "next_button": "Next"
  },
  "custom_variables": {
    "name": "label"
  },
  "footer": true,
  "folder_id":"1234",
  "preview": "https://www.surveymonkey.com/r/Preview/",
  "edit_url": "https://www.surveymonkey.com/create/",
  "collect_url": "https://www.surveymonkey.com/collect/list",
  "analyze_url": "https://www.surveymonkey.com/analyze/",
  "summary_url": "https://www.surveymonkey.com/summary/"
}
````

it has the `id` field which seems to be the only thing we use for slices, and the `date_modified` parameter. Why are we using the SurveyDetails endpoint? See https://developer.surveymonkey.com/api/v3/#surveys

Reply:
the `surveys ` itself return the small response, that contains only fields `[id, title, nickname, href]` - as well real response as well documentation says this (central part of the page). The example at right side is inactual or outdated . It returns the data wroted above ONLY when hitting https://developer.surveymonkey.com/api/v3/#surveys-id either https://developer.surveymonkey.com/api/v3/#surveys-id-details .

Reply:
That totally makes sense - thanks for the clarification.